src/items.rs

```diff
@@ -225,10 +227,7 @@ impl<'a> FmtVisitor<'a> {
         }
 
         // A conservative estimation, to goal is to be over all parens in generics
-        let args_start = generics.ty_params
-                                 .last()
-                                 .map(|tp| end_typaram(tp))
-                                 .unwrap_or(span.lo);
+        let args_start = generics.ty_params.last().map(|tp| end_typaram(tp)).unwrap_or(span.lo);
```

Comment:
I guess you need to undo some of the formatting you did before adding the mutli-line heuristic in order to preserve the multi-line-ness here (and elsewhere).